base/server/python/pki/server/__init__.py

```diff
@@ -719,10 +719,9 @@ def split_cert_id(cert_id):
     @staticmethod
     def setup_password_authentication(username, password, subsystem_name='ca', secure_port='8443'):
         """Return a PKIConnection, logged in using username and password."""
-        connection = client.PKIConnection(
-            'https', os.environ['HOSTNAME'], secure_port, subsystem_name)
+        connection = client.PKIConnection('https', os.environ['HOSTNAME'], secure_port)
         connection.authenticate(username, password)
-        account_client = pki.account.AccountClient(connection)
+        account_client = pki.account.AccountClient(connection, subsystem=subsystem_name)
         account_client.login()
         return connection
 
```

Comment:
I'm not able to add to the right line, hence adding this comment. 

Though the code handles legacy version, we might need to update the `setup_cert_authentication()` and `renew_certificate()` methods too in this file

Reply:
Yeah, let me do that in a separate PR.